humanlayer-wui/src/components/internal/SessionDetail/DangerouslySkipPermissionsDialog.tsx

```diff
@@ -0,0 +1,133 @@
+import { FC, useState } from 'react'
+import * as React from 'react'
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from '@/components/ui/dialog'
+import { Button } from '@/components/ui/button'
+import { Input } from '@/components/ui/input'
+import { Label } from '@/components/ui/label'
+import { Checkbox } from '@/components/ui/checkbox'
+import { AlertTriangle } from 'lucide-react'
+
+interface DangerouslySkipPermissionsDialogProps {
+  open: boolean
+  onOpenChange: (open: boolean) => void
+  onConfirm: (timeoutMinutes: number | null) => void
+}
+
+export const DangerouslySkipPermissionsDialog: FC<DangerouslySkipPermissionsDialogProps> = ({
+  open,
+  onOpenChange,
+  onConfirm,
+}) => {
+  const [timeoutMinutes, setTimeoutMinutes] = useState<number | ''>(15)
+  const [useTimeout, setUseTimeout] = useState(true)
+
+  // Reset to default when dialog opens
+  React.useEffect(() => {
+    if (open) {
+      setTimeoutMinutes(15)
+      setUseTimeout(true)
+    }
+  }, [open])
+
+  const handleConfirm = () => {
+    const minutes = useTimeout ? (timeoutMinutes === '' ? 15 : timeoutMinutes) : null
+    onConfirm(minutes)
+    onOpenChange(false)
+  }
+
+  return (
+    <Dialog open={open} onOpenChange={onOpenChange}>
+      <DialogContent className="sm:max-w-[475px]">
+        <DialogHeader>
+          <DialogTitle className="flex items-center gap-2 text-[var(--terminal-error)] text-base font-bold">
+            <AlertTriangle className="h-4 w-4" />
+            Bypass Permissions
+          </DialogTitle>
+          <DialogDescription asChild>
+            <div className="space-y-2">
+              <p>
+                Bypassing permissions will <strong>automatically accept ALL tool calls</strong> without
+                your approval. This includes:
+              </p>
+              <ul className="list-disc list-inside space-y-1">
+                <li>File edits and writes</li>
+                <li>Running bash commands</li>
+                <li>Reading files</li>
+                <li>Spawning sub-tasks</li>
+                <li>All MCP tool calls</li>
+              </ul>
+              <p className="text-[var(--terminal-error)] font-semibold">Use with extreme caution!</p>
+            </div>
+          </DialogDescription>
+        </DialogHeader>
+        <div className="pb-0 pt-6">
+          <div className="flex items-center justify-end space-x-2 min-h-[36px]">
+            <Checkbox
+              id="use-timeout"
+              checked={useTimeout}
+              onCheckedChange={(checked: boolean) => {
+                setUseTimeout(checked)
+                if (checked) {
+                  setTimeout(() => {
+                    const input = document.getElementById('timeout') as HTMLInputElement
```

Comment:
We can use React's `ref` abilities for this, that way you don't have to worry about grabbing the element by `id`: https://react.dev/reference/react/useRef#manipulating-the-dom-with-a-ref